platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -165,8 +167,16 @@ function WaitForDevice() {
     while [[ ("${QUERY_RETRY_COUNT}" -lt "${QUERY_RETRY_COUNT_MAX}") && ("$?" -ne "${EXIT_SUCCESS}") ]]; do
         sleep 1s
         ((QUERY_RETRY_COUNT++))
-        ${QUERY_CMD} > /dev/null
+        output=$(eval ${MFT_DIAGNOSIS_FLAGS} ${QUERY_CMD}) > /dev/null
```

Comment:
>eval [](http://example.com/codeflow?start=17&length=4)

`eval` is dangerous, even this case there is no malicious content in variables. Could you try prevent `eval` usage?

Reply:
Hi
We use it because we need to execute the environment variables along with the command. Without it, the environment variable can be treated as command, resulting in the following error
```
Fri 25 Aug 2023 10:46:38 AM IDT Rebooting with platform x86_64-mlnx_msn3420-r0 specific tool ...
/usr/bin/mlnx-fw-upgrade.sh: line 153: FLASH_ACCESS_DEBUG=1: command not found
/usr/bin/mlnx-fw-upgrade.sh: line 158: FLASH_ACCESS_DEBUG=1: command not found

```